File: dart/bubble_sort.cpp
int *bubble_sort(int arr[], int size)
{
  bool no_swaps;
  for (int i = 0; i < size; i++)
  {
    no_swaps = false;
    for (int j = 0; j < size; j++)
    {
      if (arr[j] > arr[j + 1])
      {
        int temp = arr[j + 1];
        arr[j + 1] = arr[j];
        arr[j] = temp;
        no_swaps = true;
      }
    }
    if (!no_swaps)
    {
      break;
    }
  }

  return arr;
}

File: kotlin/bubble_sort.kt
fun bubbleSort(arr: Array<Int>): Array<Int> {
  var i = 0
  var noSwaps = false
  while (i < arr.size && !noSwaps) {
    var j = 0
    noSwaps = true
    while (j < arr.size - 1) {
      if (arr[j] > arr[j + 1]) {
        var temp = arr[j + 1]
        arr[j + 1] = arr[j]
        arr[j] = temp
        noSwaps = false
      }
      j++
    }
    i++
  }
  return arr
}